### DISCUSSION TITLE
Strongswan Performance

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#1957] [General]
Does strongswan work with same performance for all kernel versions? I work with old kernel version in my processor. performance of strongswan 5.6.0 and 6.0 versions does not comply under high traffic. At the beginning, strongswan 6.0 tunnel establishment fast as in strongswan 5.6.0 but later CPU reaches ~%97 and speed of ipsec traffic gets stuck.

**Comment by tobiasbrunner** [comment]
What performance are you referring to exactly? The last part seems to indicate ESP performance, and that can definitely depend on the kernel version (and its configuration) and the negotiated algorithms. The latter might also depend on the strongSwan version as the defaults have changed over time. But please clarify what exactly you are observing.

**Reply by keskinadem** [reply]
I have 2000 peer. I mean tunnel establishment. I use AH. Whenever ı start traffik all peers want to establish IKE tunnel with center device. At the beginning, tunnel establishment speed seems normal but then gradually slow down. Established IKE tunnels make ipsec but as the CPU gets higher, ipsec traffic also slow down, even stops at some point. After all ike tunnels established, CPU become normal and ipsec traffic increase as it is expected.

**Reply by tobiasbrunner** [reply]
So what's using the CPU? What algorithms are you using? Have you read the [docs about IKE SA lookups](https://docs.strongswan.org/docs/5.9/config/lookupTuning.html) and [job priorities and IKE_SA_INIT dropping](https://docs.strongswan.org/docs/5.9/config/jobPriority.html)? You mentioned different strongSwan versions, what difference did you observe there? Did you use the same configs?

**Comment by keskinadem** [comment]
Algorithms an configurations are completely same in both version.

**Comment by keskinadem** [comment]
This might be because of XFRM interces added in version 5.8.0. (I see files src/libcharon/plugins/kernel_netlink/kernel_netlink_xfrm.c and src/xfrmi) My kernel version does not support xfrmi and can not create xfrm interface. "received netlink error: Operation not supported (95)" in line  87 /src/xfrm/xfrmi.c (manager = kernel_netlink_xfrmi_create()) Then in run time send ack keeps failing on socket xfrm.

I check documentation and see [route-based ipsec](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html) but ı do policy based. How to disable route-based? or What do you suggest?

**Reply by tobiasbrunner** [reply]
No, that has absolutely nothing to do with it. That interface is created and immediately deleted just to check if the feature is supported.
